platforms/ucp/modes/ucp-no-ads.mode.ts

```diff
@@ -1,11 +1,43 @@
-import { NoAdsMode } from '@platforms/shared';
+import { NoAdsMode, PageTracker } from '@platforms/shared';
+import { context, utils } from '@wikia/ad-engine';
 import { Injectable } from '@wikia/dependency-injection';
 import { Communicator } from '@wikia/post-quecast';
 
 @Injectable()
 export class UcpNoAdsMode implements NoAdsMode {
 	handleNoAds(): void {
 		this.dispatchJWPlayerSetupAction();
+		this.trackAdEngineStatus();
+	}
+
+	private trackAdEngineStatus(): void {
+		PageTracker.trackProp('adengine', `off_${this.getReasonForNoAds()}`);
+	}
+
+	private getReasonForNoAds(): string {
+		const reasonFromBackend = window.ads.context.opts.noAdsReason || null;
+		const pageType = window.ads.context.opts.pageType || null;
+
+		if (reasonFromBackend === 'no_ads_user' && pageType === 'homepage_logged') {
+			return null;
+		}
+
+		if (reasonFromBackend !== null) {
```

Comment:
Then this would simplify to 
```suggestion
		if (reasonFromBackend) {
```